pkg/apis/hive/v1/agent/platform.go

```diff
@@ -0,0 +1,149 @@
+package agent
+
+import (
+	//corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Platform stores cluster configuration for an agent based install where hosts are booted and registered, ready
+// to be assigned and used in a cluster.
+type Platform struct {
+
+	// Provider contains settings for different providers for which an agent based cluster installation
+	// can be performed.
+	Provider Provider `json:"provider"`
+
+	// Networking is the configuration for the pod network provider in
+	// the cluster.
+	// TODO: should this be optional and defaulted?
+	Networking Networking `json:"networking"`
+
+	// UserManagedNetworking indicates if the networking is managed by the user.
+	// TODO: more detail needed here.
+	// +optional
+	UserManagedNetworking bool `json:"userManagedNetworking,omitempty"`
```

Comment:
I found https://github.com/openshift/assisted-service/pull/674 and https://issues.redhat.com/browse/MGMT-2752 but there's not a lot of discussion in those about the "why"

As best I can make out, there effect of this is to use the 'none' platform and not require a machine network CIDR to be specified. Also that the API and Ingress VIP is not required?

This seems to be for 2 specific use cases - (a) remote worker nodes deployments where the worker nodes are not on the same L2 network as the control plane nodes, and (b) single-node deployments

Hard to see any of this as being specific to the agent-based platform

Reply:
Property was removed.